spec-cordova/prepare.spec.js

```diff
@@ -17,175 +17,265 @@
+                    expect(HooksRunner.prototype.fire.calls.count()).toBe(2);
+                    done();
+                });
+            });
         });
     });
 
-    describe('success', function() {
-        it('Test 003 : should run inside a Cordova-based project by calling util.isCordova', function(done) {
-            prepare().then(function() {
-                expect(is_cordova).toHaveBeenCalled();
-            }, function(err) {
+    describe('preparePlatforms helper method', function () {
+        let cfg_parser_mock = function () {};
+        let cfg_parser_revert_mock;
+        let platform_munger_mock = function () {};
+        let platform_munger_revert_mock;
+        let platform_munger_save_mock;
+        beforeEach(function () {
+            spyOn(prepare, 'restoreMissingPluginsForPlatform').and.returnValue(Q());
+            //cfg_parser_mock.prototype = jasmine.createSpyObj('config parser prototype mock', ['blah']);
+            //cfg_parser_mock.prototype = {};
+            cfg_parser_revert_mock = prepare.__set__('ConfigParser', cfg_parser_mock);
+            platform_munger_save_mock = jasmine.createSpy('platform munger save mock');
+            platform_munger_mock.prototype = jasmine.createSpyObj('platform munger prototype mock', ['add_config_changes']);
+            platform_munger_mock.prototype.add_config_changes.and.returnValue({
+                save_all: platform_munger_save_mock
+            });
+            platform_munger_revert_mock = prepare.__set__('PlatformMunger', platform_munger_mock);
+            spyOn(util, 'projectConfig').and.returnValue(project_dir);
+            spyOn(util, 'projectWww').and.returnValue(path.join(project_dir, 'www'));
+            
+        });
+        afterEach(function () {
+            cfg_parser_revert_mock();
+            platform_munger_revert_mock();
+        });
+        it('should call restoreMissingPluginsForPlatform', function(done) {
+            prepare.preparePlatforms(['android'],project_dir, {}).then(function() {
+                expect(prepare.restoreMissingPluginsForPlatform).toHaveBeenCalled();
+                done();
+            }).fail(function(err) {
                 expect(err).toBeUndefined();
-            }).fin(done);
+                done();
+            });
         });
-        it('Test 004 : should get PlatformApi instance for each platform and invoke its\' run method', function(done) {
-            prepare().then(function() {
-                supported_platforms.forEach(function(p) {
-                    expect(parsers[p]).toHaveBeenCalled();
-                    expect(getPlatformApi).toHaveBeenCalledWith(p);
-                });
-                expect(platformApi.run).toHaveBeenCalled();
-            }, function(err) {
+        it('should retrieve the platform API via getPlatformApi per platform provided, and invoke the prepare method from that API', function(done) {
```

Comment:
I think in this test we also need to assert that the platform API object's `prepare` method gets invoked.